### ISSUE TITLE
Wrong Avatar Image Displayed

### ISSUE THREAD

**Issue by baiti** [cgeo/cgeo#11661]
### Describe your problem!

Not sure whether this has been reported before. If it has, I need to learn how to efficiently search the issues database for keywords such as "avatar" in this case.

If I activate the GC connector and authorize cgeo for a user, the avatar of this user is shown on the right side next to "Update or remove authorization".

If I now click on exactly that text, a dialog comes up which allows me to log in a different user.  At the end the new user is authorized, the text mentiones the new user's userid, however, the avatar image is not replaced, it still renders the image of the previously logged in user.

### How to reproduce?

See above

### Reproducible

Yes

### c:geo Version

2021.09.08

**Comment by fm-sys** [comment]
In #10928, I have nearly completely refactored the handling of Avatar Images, so it is likely that it will be implicitly fixed. I will assign this to me for further testing...

**Comment by baiti** [comment]
> In #10928, I have nearly completely refactored the handling of Avatar Images, so it is likely that it will be implicitly fixed. I will assign this to me for further testing...

I believe it isn't directly related to avatar handling per se. When I did what is now known as the AL connector (fka LC connector) I stumpled over a similar situation and I don't remember who, (I believe it was actually you), recommended to do some sort of a reload at the end of that sequence.

Backing out of that view and entering it again fixes the problem already today.

**Comment by baiti** [comment]
I think it was `setResult(RESTART_NEEDED);`

**Comment by fm-sys** [comment]
> I think it was `setResult(RESTART_NEEDED);`

This will restart the whole application, but we probably just want to refresh the setting page. Anyway, I will take a look as time permits...

**Comment by baiti** [comment]
> > I think it was `setResult(RESTART_NEEDED);`
> 
> This will restart the whole application, but we probably just want to refresh the setting page. Anyway, I will take a look as time permits...

Yes, you are right and in fact from what I see it wouldn't even solve the issue. If I log in the new user and go into the check authorization and then the authorization succesds, the view closes and the old view has the new userid in the text are and only the avatar is the old avatar. So that tells me, at least part of that view got refreshed just the image isn't refreshed

**Comment by baiti** [comment]
one more thing ... fixing this isn't really of prime importance. How likely is it that a user switches accounts on a regular basis?

**Comment by Lineflyer** [comment]
I remember having seen this...but AFAIR it was a timing issue for me, meaning: The old avatar was shown until (in background) the new avatar was loaded and stored.
